### PR TITLE
Fixing

### PR DESCRIPTION
**Pull request by rwisecar**

### CODE REVIEW

File: src/linked_list.py [hunk 1]
```diff
@@ -50,17 +51,16 @@ def pop(self):
...
+            while current:
                 if current.value == val:
-                    found = True
-                    return current
+                    break
                 else:
                     current = current.next_node
-            if current is None:
+            if current:
+                return current.value
```
Comment (WillWeatherford) [inline src/linked_list.py]: Per the assignment specs, `search` should return a Node instance, not the value. Kind of weird, but that's what the specs say.

File: src/linked_list.py [hunk 2]
```diff
@@ -72,21 +72,21 @@ def display(self):
         linked_list_string += ")"
         return linked_list_string
 
-    def remove(self, node):
+    def remove(self, value):
```
Comment (WillWeatherford) [inline src/linked_list.py]: Unfortunately, the assignment specs say that `remove` should accept a node argument, rather than a value. I think that's pretty silly personally, but that's what the specs say (and also what I test for).

Also, make sure to consider testing for the edge case where someone tries to remove a node which is not in your linkedlist.

File: src/linked_list.py [hunk 3]
```diff
@@ -50,17 +51,16 @@ def pop(self):
...
                     current = current.next_node
-            if current is None:
+            if current:
+                return current.value
+            else:
                 raise ValueError("Value not found in list.")
-        return None
 
     def display(self):
         """Input a linked list and return a string showing list in tuples."""
```
Comment (WillWeatherford) [inline src/linked_list.py]: You'll need to fix this method. You're calling `str` on your node instances. However, you haven't implemented a `__str__` method on your class, which means that `str(one_of_your_nodes)` will just return the default string representation of a class, which looks like: `<linked_list.Node object at 0x7f2e0d285cf8>`

So either implement a `__str__` method, or get the string representation of the node's value.
